### ISSUE TITLE
Understadning the NPT worflow

### ISSUE THREAD

**Issue by mem48** [nptscot/npt#319]
![npt_workflow](https://github.com/nptscot/npt/assets/15819577/8f805a9a-40a5-42c9-ac47-b49f143cd6f5)

**Comment by mem48** [comment]
@joeytalbot  and @Robinlovelace see image for how the npt workflow should work.

The key bit is the six targets in the blue box. This is one functional unit of routing a journey purpose and route type (e.g. commute fastest) There are then 12 functional units in the build (4x route types and 2x purposes).

The orange box is the targets that brings all the units together into the final outputs. 

The red box shows another functional unit (schools_ebike) and the dependency between them so that only one routing target can run at a time. Buth other non-routing targets can run in parallel.

**Comment by mem48** [comment]
Also not that the balanced routing is not used in the combined outputs, so it happens last and can happen while the combined results are being built. This reduces build time a lot.

**Comment by mem48** [comment]
The code is grouped differently, usually all the targets of a particular type are together e.g. 

```
# Commute RNets -----------------------------------------------------------

tar_target(rnet_commute_fastest, {
  stplanr::overline2(uptake_commute_fastest, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
}),

tar_target(rnet_commute_quietest, {
  stplanr::overline2(uptake_commute_quietest, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
}),

tar_target(rnet_commute_ebike, {
  stplanr::overline2(uptake_commute_ebike, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
}),

tar_target(rnet_commute_balanced, {
  stplanr::overline2(uptake_commute_balanced, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
}),


```

**Comment by joeytalbot** [comment]
> The red box shows another functional unit (schools_ebike) and the dependency between them so that only one routing target can run at a time. Buth other non-routing targets can run in parallel.

Are the schools done first purely to avoid trying to route two different purposes at the same time?

**Comment by joeytalbot** [comment]
> The code is grouped differently, usually all the targets of a particular type are together e.g.
> 
> ```
> # Commute RNets -----------------------------------------------------------
> 
> tar_target(rnet_commute_fastest, {
>   stplanr::overline2(uptake_commute_fastest, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
> }),
> 
> tar_target(rnet_commute_quietest, {
>   stplanr::overline2(uptake_commute_quietest, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
> }),
> 
> tar_target(rnet_commute_ebike, {
>   stplanr::overline2(uptake_commute_ebike, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
> }),
> 
> tar_target(rnet_commute_balanced, {
>   stplanr::overline2(uptake_commute_balanced, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
> }),
> ```

In this code, we have multiple instances of `"bicycle_ebike"`. how does that work?

**Comment by joeytalbot** [comment]
Does commute fastest route at the same time as commute quietest/ebike?

**Comment by Robinlovelace** [comment]
> Does commute fastest route at the same time as commute quietest/ebike?

Yes they can, unless you make the other network types happen after fastest.

**Comment by mem48** [comment]
> > The red box shows another functional unit (schools_ebike) and the dependency between them so that only one routing target can run at a time. Buth other non-routing targets can run in parallel.
> 
> Are the schools done first purely to avoid trying to route two different purposes at the same time? 

Yes, its to only have one routing job at a time sent to cyclestreets. The schools are smaller and so are ready for routing first.

**Comment by mem48** [comment]
> > The code is grouped differently, usually all the targets of a particular type are together e.g.
> > 
> > ```
> > # Commute RNets -----------------------------------------------------------
> > 
> > tar_target(rnet_commute_fastest, {
> >   stplanr::overline2(uptake_commute_fastest, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
> > }),
> > 
> > tar_target(rnet_commute_quietest, {
> >   stplanr::overline2(uptake_commute_quietest, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
> > }),
> > 
> > tar_target(rnet_commute_ebike, {
> >   stplanr::overline2(uptake_commute_ebike, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
> > }),
> > 
> > tar_target(rnet_commute_balanced, {
> >   stplanr::overline2(uptake_commute_balanced, c("bicycle","bicycle_go_dutch","bicycle_ebike"))
> > }),
> > ```
> 
> In this code, we have multiple instances of `"bicycle_ebike"`. how does that work?

These are ebike scenario rather than ebike routing type. It is a bit confusing.

**Comment by mem48** [comment]
> > Does commute fastest route at the same time as commute quietest/ebike?
> 
> Yes they can, unless you make the other network types happen after fastest.

No only one routing job is run at a time. Each purpose and route type has its own group of targets.

**Comment by Robinlovelace** [comment]
This is v. handy Malcolm, great explanation of the build process.

**Comment by Robinlovelace** [comment]
Closing for now. Do you have any specific actions related to this? Issues are best when they have a clearly defined outcome and set of steps needed for them to be closed which I don't think we have  in this case.

Understanding of the workflow as set out here contributes to #35.

Happy to re-open if we can be more specific about the next steps on this, maybe a new issue split out from #35 or renaming this one to 'integrate other trips into build process' would make more sense.

**Comment by Robinlovelace** [comment]
#268
